tool/tctl/common/auth_command.go

```diff
@@ -504,13 +508,80 @@ func (a *AuthCommand) GenerateCRLForCA(ctx context.Context, clusterAPI authComma
 	if err := certType.Check(); err != nil {
 		return trace.Wrap(err)
 	}
-
-	crl, err := clusterAPI.GenerateCertAuthorityCRL(ctx, certType)
+	clusterName, err := clusterAPI.GetClusterName(ctx)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	authority, err := clusterAPI.GetCertAuthority(ctx, types.CertAuthID{
+		Type:       certType,
+		DomainName: clusterName.GetClusterName(),
+	}, false)
 	if err != nil {
 		return trace.Wrap(err)
 	}
 
-	fmt.Println(string(crl))
+	tlsKeys := authority.GetActiveKeys().TLS
+	if len(tlsKeys) == 0 {
+		return trace.BadParameter("CA has no active keys")
+	}
+
+	if a.output == "" {
+		if len(tlsKeys) > 1 {
+			return trace.BadParameter("CA has multiple active keys, use --out to export all CRLs")
+		}
+		crl := tlsKeys[0].CRL
+		if len(crl) == 0 {
+			fmt.Fprintf(os.Stderr, "keypair is missing CRL for %v authority %v, generating legacy fallback", authority.GetType(), authority.GetName())
+			crl, err = clusterAPI.GenerateCertAuthorityCRL(ctx, certType)
+			if err != nil {
+				return trace.Wrap(err)
+			}
+		}
+		fmt.Println(string(crl))
```

Comment:
I know this code already existed, but does it actually work? The CRL is binary data, won't adding a `\n` to it corrupt the CRL?